File: basics/10-first-non-repeated/Solution.java
import java.util.*;

class Solution {
    public char firstNonRepeated(String s) {
        Map<Character, Integer> count = new LinkedHashMap<>();
        for (char c : s.toCharArray()) {
            count.put(c, count.getOrDefault(c, 0) + 1);
        }
        for (Map.Entry<Character, Integer> e : count.entrySet()) {
            if (e.getValue() == 1) return e.getKey();
        }
        return '\0';
    }
}

File: basics/10-first-non-repeated/Solution.cpp
#include <string>
#include <unordered_map>

char firstNonRepeated(const std::string& s) {
    std::unordered_map<char, int> count;
    for (char c : s) count[c]++;
    for (char c : s) {
        if (count[c] == 1) return c;
    }
    return '\0';
}
